### PR TITLE
genpolicy: add support for cc-azurefile-csi driver

### PR DESCRIPTION
**Pull request by arc9693**
This patch adds support for the cc-azurefile-csi driver to the genpolicy.

###### Merge Checklist  

- [ ] Followed patch format from upstream recommendation: https://github.com/kata-containers/community/blob/main/CONTRIBUTING.md#patch-format
  - [ ] Included a single commit in a given PR - at least unless there are related commits and each makes sense as a change on its own.
- [x] Aware about the PR to be merged using "create a merge commit" rather than "squash and merge" (or similar)
- [x] genPolicy only: Ensured the tool still builds on Windows
- [x] genPolicy only: Updated sample YAMLs' policy annotations, if applicable
- [x] The `upstream-missing` label (or `upstream-not-needed`) has been set on the PR.

### CODE REVIEW

**Review by danmihai1 — APPROVED** [review]
Looks like a great start! Please add Policy rules for validating the SMB mount flags (e.g., execution not allowed, etc.)

**Review by ms-mahuber — APPROVED** [review]

**Comment by Redent0r** [comment]
Verified it build on Windows and started test run https://dev.azure.com/mariner-org/mariner/_build/results?buildId=556262&view=results

**Review by sprt — CHANGES_REQUESTED** [review]
Could you update the samples using this script? https://github.com/microsoft/kata-containers/blob/msft-main/src/tools/genpolicy/update_policy_samples.py

File: src/tools/genpolicy/rules.rego [hunk 1]
```diff
@@ -912,10 +912,18 @@ allow_storage_options(p_storage, i_storage, layer_ids, root_hashes) {
 
     p_storage.driver == "smb"
     count(p_storage.options) == 0
```
Comment (sprt) [inline src/tools/genpolicy/rules.rego:914]: Is this not redundant assuming we trust genpolicy?

**Review by sprt — APPROVED** [review]
I would squash commits \#1 and \#2 but LGTM!

**Review by Redent0r — APPROVED** [review]

File: src/tools/genpolicy/rules.rego (hunk 1 above)
Comment (danmihai1) [reply src/tools/genpolicy/rules.rego:914]: I agree we don't need to check the count of policy arrays.

**Review by danmihai1 — APPROVED** [review]

**Comment by arc9693** [comment]
Test rerun: https://dev.azure.com/mariner-org/mariner/_build/results?buildId=556788&view=results

**Pull request merged**